### PR TITLE
fix: automatically set default schema env var for type generation

### PR DESCRIPTION
**Pull request by bug-author**
## What kind of change does this PR introduce?

This PR lets the CLI automatically detect when exactly one non‑public schema is provided and sets the environment variable `PG_META_GENERATE_TYPES_DEFAULT_SCHEMA` accordingly.

## What is the current behavior?

Currently, when running `supabase gen types`, the CLI passes the schema flags directly to `postgres‑meta` without automatically setting a default schema. This results in the generated TypeScript file always defaulting to the "public" alias:

```ts
export type PublicSchema = Database[Extract<keyof Database, "public">]
```

Even if a non‑public schema is specified (for example, `--schema robo`), helper types (like `Tables<"test_table">`) incorrectly reference the public schema. 

Relevant issues:
#3061 
#3146 

## What is the new behavior?

The CLI now automatically checks if exactly one schema is provided (and if that schema is not "public"). If so, it sets the environment variable `PG_META_GENERATE_TYPES_DEFAULT_SCHEMA` to that schema's name. This allows `postgres‑meta` to generate a default alias based on the provided schema (for example, for "robo", the generated file will include):

```ts
export type DefaultSchema = Database[Extract<keyof Database, "robo">]
export type DefaultSchemaOrPublic = DefaultSchema
```

In single‑schema mode, shorthand type references (like `Tables<"test_table">`) correctly resolve to the non‑public schema. In multi‑schema scenarios, the variable is not set, and users must explicitly specify the schema for non‑public tables.

## Additional context

Corresponding [postgres-meta PR](https://github.com/supabase/postgres-meta/pull/904)

### CODE REVIEW

File: internal/gen/types/types.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ func Run(ctx context.Context, projectId string, dbConfig pgconn.Config, lang str
 	}
 	included := strings.Join(schemas, ",")
 
+	var defaultSchemaEnv string
+	if len(schemas) == 1 && schemas[0] != "public" {
+		defaultSchemaEnv = schemas[0]
+	}
```
Comment (sweatybridge) [inline internal/gen/types/types.go:41]: We should add a new flag so it doesn't break existing users that rely on the generated default schema.

Comment (bug-author) [reply internal/gen/types/types.go:41]: @sweatybridge Thanks. I've implemented a `set-default` flag for backwards compatibility that makes the behavior now explicit and opt-in. The new behavior is now:

```sh
supabase gen types --schema private --set-default
```

**Comment by bug-author** [comment]
I've added 4 tests as well to test the core feature, backward compatibility and edge cases.
